test/functional/test-viewer.js

```diff
@@ -1499,135 +1499,6 @@ describe('Viewer', () => {
       const viewer = new Viewer(ampdoc);
       expect(viewer.getResolvedViewerUrl()).to.equal('https://acme.org/doc1');
     });
-
-    it('should always return current location for top-level window', () => {
```

Comment:
It'd seem some of these tests are still valid...

Reply:
Good question. I think you can just replace `isTrustedViewer()` in the tests and only test the `getResolvedViewerUrl()`?

Reply:
So Implemented this change, and noticed the following code within the `this.viewerUrl_` promise: https://github.com/ampproject/amphtml/blob/master/src/service/viewer-impl.js#L425

Meaning, the resolved URL is not updated without calling the promise that is returned by `viewer.getViewerUrl()` which is simply `this.viewerUrl_`.

I tried replacing with `isTrustedViewer()`, but that wont update the `this.resolvedViewerUrl_`, therefore the tests will break because the viewer url never actually updates.  

Perhaps this code is not entirely dead as mentioned in #15425 ? Or we need to update the `viewer-impl.js` accordingly to update it's resolved url.

Reply:
Yeah. Please consult @lannka if that code is indeed dead. Obviously some part of it is still in use.